_posts/2021-12-28-trino-2021-a-year-of-growth.md

```diff
@@ -0,0 +1,347 @@
+development on the project?
+
+At the start of the year, there were 21,924 commits. This year, we pushed 3,679 
+commits to the repository,sitting at over 25,600 now. Looking at the graph, this
+keeps us pretty consistent with 2020s throughput.
+
+<p align="center">
+ <img align="center" width="75%" src="/assets/blog/2021-review/commits.png"/>
+</p>
+
+With the project's trajectory displayed in numbers, let's examine the top 
+features that landed in Trino this year.
+
+## Features
+
+Here's a high-level list of the most exciting features that made their way into
+Trino in 2021. For details and to keep up you can check out the [release notes]({{site.url}}/docs/current/release.html).
+
+### Improvements to ANSI SQL support
+
+SQL language support is crucial for the increasing complexities of queries and 
+usage of Trino. In 2021 we added numerous new language features and 
+improvements:
+
+* [`RANGE`]({{site.url}}/blog/2021/03/10/introducing-new-window-features.html#new%20features)
+keyword for usage within a `WINDOW` function.
+* [`MATCH_RECOGNIZE`]({{site.url}}/blog/2021/05/19/row_pattern_matching.html)
+a feature that allows for complex analysis across multiple rows. To learn more 
+about this feature watch [the Community Broadcast show](/episodes/23.html).
+* Aggregate functions in for `MATCH_RECOGNIZE`.
+* `TRUNCATE TABLE` which allows for a more efficient delete.
+* `DENY` statement, which enables you to remove a user or groups access via SQL.
+* `IN catalog` clause to `CREATE ROLE`, `DROP ROLE`, `GRANT ROLE`, 
+`REVOKE ROLE`, and `SET ROLE` to specify the target catalog of the statement 
+instead of using the current session catalog.
+* [`listagg()`]({{site.url}}/docs/current/functions/aggregate.html#listagg)
+function returns concatenated values seperated by a specified separator.
+* [`SET TIME ZONE`]({{site.url}}/docs/current/sql/set-time-zone.html) keywords.
+* Arbitrary queries in [`SHOW STATS`]({{site.url}}/docs/current/sql/show-stats.html).
+* Subquery expressions that return multiple columns. Example: `SELECT x = (VALUES (1, 'a'))`.
+* Add support for `ALTER MATERIALIZED VIEW` ... `RENAME TO` ...
+
+### Query processing improvements
+
+* Added support for automatic query retries.
+* Updated the behavior of `ROW` to `JSON` cast to produce `JSON` objects instead
+of `JSON` arrays.  
+
+## Performance improvements
+
+Improved performance for the following operations and processing:
+
+* Querying Parquet data for files containing column indexes.
+* Reading dictionary-encoded Parquet files.
+* Queries using [`rank()`]({{site.url}}/docs/current/functions/window.html#rank) `WINDOW` function.
+* Queries using [`sum()`]({{site.url}}/docs/current/functions/aggregate.html#sum)
+and [`avg()`]({{site.url}}/docs/current/functions/aggregate.html#avg) for 
+decimal types.
+* Queries using `GROUP BY` with single grouping column.
+* Speed of decimal aggregations.
+* Turnaround for evaluation of the WHERE and SELECT clause.
```

Comment:
What does this mean?

Reply:
I tried another way to say "Improved..."